src/storage/heap_file.c

```diff
@@ -1233,6 +1233,8 @@ heap_scan_pb_lock_and_fetch_debug (THREAD_ENTRY * thread_p, const VPID * vpid_pt
   LOCK page_lock;
   PGBUF_LATCH_MODE page_latch_mode;
 
+  assert (lock == S_LOCK || lock == X_LOCK);
```

Comment:
>  lock can be only S_LOCK or X_LOCK since all the callers of this function gives S_LOCK or X_LOCK as parameter lock.

This pre-condition could be true, but I don't think this can be the reason. If others are not used doesn't mean that it's not allowed. It could be that there is just no use case. We need another proper reason to go with this.

Reply:
@wrlawodms That's a good comment. However, verification of your request is beyond the scope of this issue. So, we have two options. One is to not edit the part you mentioned, and the other is to leave a comment with relevant information in the edited part. I think it would be a good idea to leave information through the second option(track it as a comment) and then check whether only S_LOCK and X_LOCK are allowed in later.

Reply:
Do we need to change the pre-condition check even without verification? It's trivial, but I don't see the benefit of this change.

Reply:
I mean, is this worth it though it even needs a comment saying "we're not sure" for this?

Reply:
This issue is not one that accomplishes any meaningful work. The purpose is to reduce minor errors by removing unnecessary code and removing unnecessary conditions. Comments will be written in TODO: format and will simply confirm whether LOCKs other than S_LOCK and X_LOCK are allowed.

Reply:
>  The purpose is to reduce minor errors by removing unnecessary code and removing unnecessary conditions

I think it's meaningful, but we don't know if the assertion is unnecessary.
If you don't want to debate, you can go ignoring my comment. :)  If you want to keep the debate, I still don't know how this is a work for the purpose you said.

Reply:
I always feel our discussion is worth. However this part is too minor and we have to use our time more productive way. So, I choose my opinion and leave a comment for this to track the issue that this function is only allowed by S_LOCK and X_LOCK. Thanks : )

Reply:
Yes, that's what I mean:)